Refactor all deployments to use the MSI package. (#3284)
microsoft/ebpf-for-windows: docs/InstallEbpf.md

### Method 3 (Install files you built yourself, with a VM checkpoint)

This method uses a machine that has already built the binaries for

>  the desired build configuration, i.e. `\x64\[Debug|Release|NativeOnlyDebug|NativeOnlyRelease]`.

Reviewer: ```suggestion
 the desired build configuration, i.e., `\x64\[Debug|Release|NativeOnlyDebug|NativeOnlyRelease]`.
```
nit

   New-StoredCredential -Target TEST_VM -Username <VM Administrator> -Password <VM Administrator account password> -Persist LocalMachine
   ```

>    > Note that "`TEST_VM`" is literal and is later used to lookup the actual VM name; it need not be the name of any actual test VM.

Reviewer: ```suggestion
   > Note that "`TEST_VM`" is literal and is later used to look up the actual VM name; it need not be the name of any actual test VM.
```
"look up" is a verb, "lookup" is a noun

> 1. Enter the desired directory (`cd`) where the build artifacts are stored (i.e. `\x64\[Debug|Release|NativeOnlyDebug|NativeOnlyRelease]`).

Reviewer: ```suggestion
1. Enter the desired directory (`cd`) where the build artifacts are stored (i.e., `\x64\[Debug|Release|NativeOnlyDebug|NativeOnlyRelease]`).
```

1. Modify `.\test_execution.json` to specify the name of the test VM under the `VMMap` attribute, e.g.:

    ```json
    {
        ...

        "VMMap":
        {

>             "TEST_VM":

Reviewer 2: this is incorrect - should be GH_RUNNER_NAME

Author: this is a snippet of the `test_execution.json`, which was not bundled with updates to this doc: in the context of this paragraph, `TEST_VM` looks correct to me, in case not, please feel free to suggest the update that should have been made. Thanks!

```xml
{
    "license":["Copyright (c) Microsoft Corporation", "SPDX-License-Identifier: MIT"],

    "VMMap":
    {
        "TK5-3WP08R0904_WS2019_1":
        [
            {
                "Name": "vm1_ws2019"
            }
        ],
        "TK5-3WP08R0904_WS2019_2":
        [
            {
                "Name": "vm2_ws2019"
            }
        ],
        ...
   }
}
```

Reviewer 3: TEST_VM provides a unique 'resource name' for storing the credentials at (It is a parameter for `New-StoredCredential`, `Get-StoredCredential` commandlets etc.). It can be anything so long as the callers use the right name to access the right creds.

(In the CI/CD scripts we use TEST_VM for storing the 'Administrator' user creds and TEST_VM_STANDARD for storing 'StandardUser' user creds.)

In this particular usage it seems to indicate a relation to the test vm name itself, when there isn't any and can be confusing.  Maybe we could use a different name in the doc, say something like "my_test_vm_name"?

Author: I think this is already the case with the current change, right?

```json
{
    ...

    "VMMap":
    {
        "TEST_VM":
        [
            {
                "Name": "<test-vm-name>"
            }
        ],
        ...
    }
    ...
}
```

Reviewer 2: no, this is not right. what you call TEST_VM, which suggests this stirng represents a user-created name for a VM, should instead be a GH runner service instance name that they will use.

Separately, to Dhiren's point, the specific string TEST_VM is the name of a stored credential we keep on the host, and we use this name elsewhere in the doc.

Author: @rectified95, these are just updated user instructions for running the CICD tests on a *local VM named "`<test-vm-name>`"*. What the `TEST_VM` literal represents was (and still is in the same way) clarified in the doc:

```
> Note that "`TEST_VM`" is literal and is later used to look up the actual VM name; it need not be the name of any actual test VM.
```

Since this update was missed when the `test_execution.json` format was updated, I'm simply adding it in this PR for simplicity, as also anticipated offline.

I used these new instructions myself to run the entire CICD locally on one of my VMs, and worked perfectly.

As mentioned earlier, you're welcome to constructively just suggest the modifications that you think should have been done when the `test_execution.json` format was updated. Thanks.

Reviewer 3: +1 for @rectified95's comment above.

It really is the github runner service instance name.  `TEST_VM` is being used for something else altogether and is especially confusing when that same string is used as a service instance name.  IMO, something like `"<my-test-runner-service-instance-name>"` would make it unambiguous.

Reviewer 3: In retrospect, checking in a test_execution.json with hard-coded names was a mistake (https://github.com/microsoft/ebpf-for-windows/issues/3347).

Author: I see this is a much bigger problem, and the initial intent of just sneaking in a small doc change is not appropriate to be done in this PR. 
As agreed offline, since updating this doc is completely unrelated to this PR, I'm pulling the changes out, and adding a comment to #3347.
